src/views/Home/components/UserBanner/HarvestCard.tsx

```diff
@@ -24,14 +24,32 @@ const HarvestCard = () => {
   const masterChefContract = useMasterchef()
   const cakePriceBusd = usePriceCakeBusd()
   const earningsBusd = new BigNumber(farmEarningsSum).multipliedBy(cakePriceBusd)
-  const numFarmsToCollect = farmsWithStakedBalance.length
+  const numTotalToCollect = farmsWithStakedBalance.length
+  const numFarmsToCollect = farmsWithStakedBalance.filter((value) => value.pid !== 0).length
+  const hasCakePoolToCollect = numTotalToCollect - numFarmsToCollect > 0
 
-  const earningsText = t('%earningsBusd% to collect from %count% %farms%', {
+  let earningsText = t('0 to collect')
+  const data = {
     earningsBusd: earningsBusd.toString(),
-    count: numFarmsToCollect > 0 ? numFarmsToCollect : '',
-    farms: numFarmsToCollect === 0 || numFarmsToCollect > 1 ? 'farms' : 'farm',
-  })
-  const [preText, toCollectText] = earningsText.split(earningsBusd.toString())
+    count: numFarmsToCollect,
+  }
+
+  if (numFarmsToCollect > 0 && hasCakePoolToCollect) {
+    if (numFarmsToCollect > 1) {
+      earningsText = t('%earningsBusd% to collect from %count% farms and CAKE pool', data)
+    } else {
+      earningsText = t('%earningsBusd% to collect from %count% farm and CAKE pool', data)
+    }
+  } else if (numFarmsToCollect > 0) {
+    if (numFarmsToCollect > 1) {
+      earningsText = t('%earningsBusd% to collect from %count% farms', data)
+    } else {
+      earningsText = t('%earningsBusd% to collect from %count% farm', data)
+    }
+  } else if (hasCakePoolToCollect) {
+    earningsText = t('%earningsBusd% to collect from CAKE pool', data)
+  }
```

Comment:
It is unfortunate our translation system doesn't have an intuitive way to handle plurals. Can you move this to a helper file or somewhere out of this component. This is quite a lot of logic to just handle text.

Reply:
Done